### PR TITLE
Wrap the Conjugate Gradient Solver Interface in Cython

### PR DESCRIPTION
**Pull request by ProfFan**
Added corresponding entries in `gtsam.h` and necessary utilities in C++.

### CODE REVIEW

**Comment by dellaert** [comment]
I am on my phone so I can’t do a proper review, but “set” Is not functional, so I don’t support it. In addition, I missing a unit test for the newly wrapped python functionality.

GitHub <notifications@github.com> wrote:
“@ProfFan requested your review on: #394 Wrap the Conjugate Gradient Solver Interface in Cython.”

**Comment by ProfFan** [comment]
@dellaert This is because `setXXX` is the convention used in all params like LMParams, etc.

File: cython/gtsam/tests/test_NonlinearOptimizer.py [hunk 1]
```diff
@@ -17,7 +17,8 @@
 import gtsam
 from gtsam import (DoglegOptimizer, DoglegParams, GaussNewtonOptimizer,
                    GaussNewtonParams, LevenbergMarquardtOptimizer,
-                   LevenbergMarquardtParams, NonlinearFactorGraph, Ordering,
+                   LevenbergMarquardtParams, PCGSolverParameters,
+                   DummyPreconditionerParameters, NonlinearFactorGraph, Ordering,
```
Comment (varunagrawal) [inline cython/gtsam/tests/test_NonlinearOptimizer.py:21]: `DummyPrecoditionerParameters` is such as weird name, but okay.

File: gtsam/linear/PCGSolver.cpp [hunk 2]
```diff
@@ -45,6 +45,17 @@ PCGSolver::PCGSolver(const PCGSolverParameters &p) {
   preconditioner_ = createPreconditioner(p.preconditioner_);
 }
 
+void PCGSolverParameters::setPreconditionerParams(const boost::shared_ptr<PreconditionerParameters> preconditioner) {
+  preconditioner_ = preconditioner;
+}
+
+void PCGSolverParameters::print(const std::string &s) const {
```
Comment (varunagrawal) [inline gtsam/linear/PCGSolver.cpp:52]: Interesting that overloading of `ostream` is missing. That just seems easier...

**Comment by ProfFan** [comment]
@varunagrawal  Sure :)

**Pull request merged**